### PR TITLE
CLDR-13071 Adding basic metadata for ars. Re-generating likelySubtags.

### PR DESCRIPTION
**Pull request by sffc**
See discussion in the CLDR bug: https://unicode-org.atlassian.net/browse/CLDR-13071

Additional context: https://github.com/unicode-org/icu/pull/660

### CODE REVIEW

**Comment by sffc** [comment]
Note: I ran `CLDR_DIR=../.. ant GenerateMaximalLocales` to re-generate likelySubtags.xml.  Running that script brought in a number of other changes, though, not just my `ars` change.  I think it's just been a little while since anyone's run that script.

**Comment by sffc** [comment]
OK.  FYI, the only change I actually made were the two lines in common/supplemental/supplementalData.xml.  The rest of the changes were created after I ran ant GenerateMaximalLocales.

**Comment by macchiati** [comment]
Closing, since the generator needs fixing. Keep the ticket open for that.

**Comment by sffc** [comment]
Can someone give a pointer to the "right" way to add *ars* to CLDR?  It is blocking work on the ICU ticket for the Ldml2IcuConverter.

**Comment by macchiati** [comment]
There are two items.
1. Add ars with population figures to at least one country where it is spoken, ideally the one with the largest usage (population*frequency of usage). That gets it into the Likely Subtags data.
2. We are planning to add a mechanism for having LocaleMatcher fallbacks from encompassed languages that to the 'predominant' encompassed language. That should be done as a code change, for all of them.

**Comment by macchiati** [comment]
... and that's what CLDR-13071 should do. Adding comments there.

**Comment by macchiati** [comment]
Now, I think you could go ahead and make 2 changes, if the tests pass afterwards.

<languageMatch desired="ars" supported="ar" percent="90" oneway="true"/>

<languagePopulation type="ars" populationPercent="0"/>	

Except for the latter; don't add languages that nobody speaks. It should look something like this line:

<languagePopulation type="gsw" writingPercent="5" populationPercent="65" officialStatus="de_facto_official" references="R1006"/>
except for the officialStatus.

**Comment by sffc** [comment]
PTAL, did I edit the write files?

I calculated the population percentage as follows:

- 843,000 ars speakers in SA in 2011 ([source](https://www.ethnologue.com/language/ars))
- 28.24M people living in SA in 2011 ([source](https://www.google.com/search?q=population+of+saudi+arabia+in+2011))

which implies 2.99% of the population of SA speaks ars.

File: common/supplemental/supplementalData.xml [hunk 1]
```diff
@@ -3885,6 +3886,7 @@ XXX Code for transations where no currency is involved
 		</territory>
 		<territory type="SA" gdp="1775000000000" literacyPercent="87.2" population="33091100">	<!--Saudi Arabia-->
 			<languagePopulation type="ar" populationPercent="100" officialStatus="official"/>	<!--Arabic-->
+			<languagePopulation type="ars" populationPercent="2.99"/>	<!--Najdi Arabic-->
```
Comment (macchiati) [inline common/supplemental/supplementalData.xml]: If the language is normally not written (that's why I gave the example of Swiss German), then we set the writingPercent to low.

File: common/supplemental/supplementalData.xml [hunk 2]
```diff
@@ -1315,6 +1315,7 @@ XXX Code for transations where no currency is involved
 		<language type="arp" scripts="Latn"/>
 		<language type="arq" scripts="Arab"/>
 		<language type="arq" territories="DZ" alt="secondary"/>
+		<language type="ars" scripts="Arab" territories="SA"/>
```
Comment (macchiati) [inline common/supplemental/supplementalData.xml]: This doesn't hurt, but note that these items are overwritten when the file is processed.

Comment (sffc) [reply common/supplemental/supplementalData.xml]: Done

File: common/supplemental/supplementalData.xml (hunk 1 above)
Comment (sffc) [reply common/supplemental/supplementalData.xml]: Done

**Comment by sffc** [comment]
I see some failing tests:

```
     [java]     testFallbacks {
     [java]       Error: (LanguageInfoTest.java:159) : fallback for ars, 90: expected com.ibm.icu.util.ULocale<ar>, got com.ibm.icu.util.ULocale<mul>
     [java]     } (0.066s) FAILED (1 failure(s), 1 warning(s))

     [java]     TestThatLanguagesHaveScript {
     [java]       Error: (LanguageTest.java:118) Language without likely script:	ars	Najdi Arabic
     [java]     } (0.021s) FAILED (1 failure(s))
```

I'll try to figure out how to fix those, but do you have any initial suggestions?  Note: the second test failure only showed up after I removed the line you suggested I remove in my latest commit.

**Comment by macchiati** [comment]
Doing both at the same time is not the normal order. Why don't you try doing the supplementalData change first, and see if that gets through the tests. Once we run the BRS Updating Codes step, then you can do the other part.

**Comment by sffc** [comment]
OK, I got the tests to pass again.  I sent an email to cldr-dev2 to accept the ticket, and I squashed the branch.

**Comment by sffc** [comment]
@pedberg-icu said in email that it is OK to merge this PR, and he will follow up with another PR to also change data in country_language_population_raw.txt.

I need @pedberg-icu and @JCEmmons to approve the PR in order to unblock merging it.  **Please merge as soon as the button turns green** since I will be OOO for a couple of weeks.

**Comment by pedberg-icu** [comment]
> Looks like the manual changes for cic were nuked. I didn't have country / language/pop data for those at the time, but they shouldn't be overwritten.

In the current diffs I do not see any changes for cic. John, can you re-review and remove your change request? Thanks.

**Pull request merged**